src/appProvider.js

```diff
@@ -15,16 +15,13 @@ import { store, history } from './redux/store';
 
 import '../src/style/styles.less';
 
-const defaultidx = localStorage.getItem('localindex') || 0;
-const locales = [AppLocale.en, AppLocale.zh];
-class AppProvider extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { locale: defaultidx };
-    this.langHandler = this.langHandler.bind(this);
+export default class AppProvider extends Component {
+  locales = [AppLocale.en, AppLocale.zh]
+  state = {
+    locale: localStorage.getItem('localindex') || 0,
   }
 
-  langHandler() {
+  langHandler = () => {
```

Comment:
is this necessary for this to work? i'd prefer to keep it as functions itself versus a variable that holds a function.

unless there is some specific benefit for doing it this way?

Reply:
It autobinds this to the class so you don't have to do it in the constructor or wherever the method is called.  It's just a fat arrow function definition, not a variable.
Old:
```javascript
class AppProvider extends Component {
  constructor(props) {
    this.state = { ... }
    this.langHandler = this.langHandler.bind(this)
  }
  langHandler() {
  }
}
```
New:
```javascript
class AppProvider extends Component {
  state = {}
  langHandler = () => {
  }
}
```

Reply:
You can see how repetitive that gets if you have like 5 methods and you have to bind all of them in the constructor or have callbacks in your render method like `() => this.langHandler`

Reply:
gotcha, this makes it much easier than to bind everything 👍